src/routes/reference/bufnmf.svx

```diff
@@ -9,4 +9,65 @@ tags:
+
+<IndentNote type='warn'>
+
+For an example of audio decomposition using NMF, see the [Decomposition with NMF](/overviews/bufnmf) Overview.
+
+</IndentNote>
+
+<Image
+src="/overviews/bufnmf/01_Bases_plot.png"
+label="Bases or 'Spectral Templates' of a decomposed drum loop. Each is a spectrum of magnitudes. (blue = kick, orange = snare, green = hi-hat)"
+/>
+
+<Image
+src="/overviews/bufnmf/00_resynth_buffer_with_activations.png"
+label="Activations from the same decomposed drum loop, overlaid on the decomposed audio components. (blue = kick, orange = snare, green = hi-hat)"
+/>
+
+NMF is a form of unsupervised machine learning: it begins either from seeded activations and bases or in a stochastic state (where the bases and activations of the components are initially random) and works iteratively, using stochastic gradient descent, trying to find a combination of components that, when summed together, minimize the difference from the original spectrum. In this implementation, the components are reconstructed by masking the original spectrum, such that they will always sum to yield the original sound.
+
+There is no single correct solution to this process. There are different ways of accounting for an spectrum in terms of some set of bases and activations. NMF tends to converge very quickly at first and then level out. Fewer iterations mean less processing, but also less predictable results.
+
+### Outputs
+
+BufNMF can return any or all of the following:
+* The bases of the components in the form of a magnitude spectrum for each.
+* The activations of the components in the form of an amplitude envelope for each.
+* An audio reconstruction of each component.
+
+### Different Modes
+
+Some additional options and flexibility can be found through combinations of the **basesMode** and **actMode** arguments.
+
+If either or both of these flags are set to 1, BufNMF expects to be supplied with the corresponding elements (activations or bases) to use as seeds for the decomposition, providing more guided results. It is possible to set both flags equal to 1.
+
+If either of these flags are set to 2, BufNMF will not modify the supplied corresponding elements (activations or bases), instead using them as templates to match against. BufNMF will modify the other elements which will represent their best match to the supplied spectrogram, given the fixed elements provided.  Note that having both basesMode and actMode set to 2 doesn't make sense.
+
+<IndentNote type='warn'>
+
+For more information on using **basesMode** and **actMode**, see the [Seeding Bases and Activations](/overviews/seeding-bases-and-activations) Overview.
+
+</IndentNote>
+
+If supplying pre-formed data (for **actMode** and **basesMode** 1 and 2), it's up to the user to make sure that the supplied buffers are the right size: bases must be **(fft size / 2) + 1** frames and **components * input channels** channels
```

Comment:
Maybe flag here that the best way to do this is to ensure you use the fftsetting when processing with the family of objects.

Reply:
is this a Max specific thing?

Reply:
Shouldn't be. Just that if you do some NMF in one place and wait to supply that data to another nmf thing later, being explicit and consistent with that parameter helps to ensure you get the right sizes rather than relying on the default (which should be the same but its probably good hygiene and practice to do otherwise).